core/tasks/model_monitoring.py

```diff
@@ -0,0 +1,191 @@
+from celery import shared_task
+import pandas as pd
+from django.db import transaction
+import django
+
+from core.models import (
+    Entry,
+    Lead,
+    Project,
+    ClassificationPredictions,
+    ClassificationModel,
+    ProjectWisePerfMetrics,
+    AllProjectPerfMetrics,
+    CategoryWiseMatchRatios,
+    ProjectWiseMatchRatios,
+    TagWisePerfMetrics,
+    ComputedFeatureDrift,
+)
+from nlp_scripts.model_monitoring.utils import get_model_info
+from nlp_scripts.model_monitoring.generate_outputs import ClassificationModelOutput
+from nlp_scripts.model_monitoring.model_performance import ModelPerformance
+from nlp_scripts.model_monitoring.featuredrift import FeatureDrift
+
+
+def save_classification_prediction(df):
+    classification_rows = df.to_dict("records")
+    model_info = get_model_info("main-model-cpu")[0]
+    model = ClassificationModel.objects.filter(
+        name=model_info["name"],
+        version=model_info["version"],
+        model_uri=model_info["model_uri"]
+    ).first()
+    print("Model : ", model)
+    if not model:
+        print("INside :***************")
+        model = ClassificationModel.objects.create(
+            name=model_info["name"],
+            version=model_info["version"],
+            model_uri=model_info["model_uri"],
+            defaults={
+                "test_description": model_info["description"],
+                "train_data_uri": model_info["reference_train_data"]
```

Comment:
None of these fields exist in the model. Also, please run the scripts again from start to the end and make sure the changes do not break anything.

Reply:
@bewakes It is running fine. we can get description and referene_train_data from get_model_info. Then save in default field as json data

Reply:
Ah! I see what's happening.

In the method `get_or_create`, the kwarg `defaults` does not represent the model field. It represents the dict of other attributes that are not used to `get` the object from the database. Please have a look here: https://docs.djangoproject.com/en/4.1/ref/models/querysets/#get-or-create

Thus, what you need to do is, remove the `defaults` field, and add the other two `description` and `refrerence_data_uri` fields in the model and use the `defaults` kwarg in `get_or_create` with `{"description": <some description>, "reference_data_uri": <some s3 url>}

I hope that makes sense.